crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -333,3 +366,127 @@ impl SlidingSyncRoom {
         Default::default()
     }
 }
+
+/// Configure Sliding-Sync extensions.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct ExtensionsRequest {
+    /// Request to devices messages with the given config.
+    pub to_device: Option<ToDeviceRequest>,
+    /// Configure the end-to-end-encryption extension.
+    pub e2ee: Option<E2EERequest>,
+    /// Configure the account data extension.
+    pub account_data: Option<AccountDataRequest>,
+    /// Extensions may add further fields to the list.
+    #[serde(flatten, default, skip_serializing_if = "BTreeMap::is_empty")]
+    pub other: BTreeMap<String, serde_json::Value>,
+}
+
+impl ExtensionsRequest {
+    fn is_empty(&self) -> bool {
+        !(self.to_device.is_some()
+            || self.e2ee.is_some()
+            || self.account_data.is_some()
+            || !self.other.is_empty())
```

Comment:
You want this legibility? I picked the other style because it was stopping early: as soon as the first would return true, we don't actually have to test the others. but I guess that is a pointless optimization here that complicates the code ...

Reply:
This one will also stop early, `&&` and `||` are both short-circuiting.